src/gt4py/next/otf/workflow.py

```diff
@@ -267,24 +287,61 @@ def skip_condition(self, inp: StartT) -> bool:
         raise NotImplementedError()
 
 
-@dataclasses.dataclass
-class InputWithArgs(Generic[StartT]):
-    data: StartT
-    args: tuple[Any]
-    kwargs: dict[str, Any]
+@dataclasses.dataclass(frozen=True)
+class DataArgsPair(Generic[DataT, ArgT]):
```

Comment:
Heads up, this will be a slightly bigger refactoring than a simple rename, because I would like to keep the data sctructures in `otf.workflow` usage-agnostic. Or in terms of another comment on this PR "named by what they are, not by the context they are used in".